---
layout: page
title: Learning Goals
menu: true
order: 5
---

By the end of this course, you will be able to: 

- Examine and extract information from online bioinformatics resources, including major biomolecular and genomic databases, search and analysis tools, genome browsers, and structure viewers.
- Select the appropriate bioinformatics tools and resources for quality control and analysis to solve complex problems in the biological sciences and medical research.
- Develop bioinformatics pipelines that collect, analyze and provide insightful visualizations for helping in the interpretation of the results obtained using publicly available sequencing data.
- Recognize the research objectives of the bioinformatics-related sub-disciplines of Genome informatics, Transcriptomics, and Structural informatics. 

In short, students will develop a solid foundational knowledge of bioinformatics and be able to evaluate new biomolecular and genomic information using state-of-the-art bioinformatic tools and resources. 

<!-- 
## Specific Learning Goals

All students who receive a passing grade should be able to:


|   |   | Lecture(s): |
|----|----|----| 
|1  | Appreciate and describe in general terms the role of computation in hypothesis-driven discovery processes within the life sciences. | 1, 2, 20 |
| 2 | Be able to query, search, compare and contrast the data contained in major bioinformatics databases and describe how these databases intersect (GenBank, GENE, UniProt, PFAM, OMIM, PDB, UCSC, ENSEMBLE). | 2, 11, 12 |
| 3 | Describe how nucleotide and protein sequence and structure data are represented (FASTA, FASTQ, GenBank, UniProt, PDB). | 2, 3, 11 |
| 4 | Be able to describe how dynamic programming works for pairwise sequence alignment and appreciate the differences between global and local alignment along with their major application areas. | 2, 3 |
| 5 | Calculate the alignment score between two nucleotide or protein sequences using a provided scoring matrix and be able to perform BLAST, PSI-BLAST, HMMER and protein structure based database searches and interpret the results in terms of the biological significance of an e-value. | 2, 3 |
| 6 | Use R to read and parse comma-separated (.csv) formatted files ready for subsequent analysis. | 4, 5, 6, 11, 13, 15, 16 |
| 7 | Perform elementary statistical analysis on biomolecular and "omics" datasets with R and produce informative graphical displays and data summaries. | 9, 10, 11, 13, 15, 16 |
| 8 | View and interpret the structural models in the PDB. | 10, 11 |
| 9 | Explain the outputs from structure prediction algorithms and small molecule docking approaches. | 11 |
| 10 | Appreciate and describe in general terms the rapid advances in sequencing technologies and the new areas of investigation that these advances have made accessible. |  13, 14, 15 |
| 11 | Understand the process by which genomes are currently sequenced and the bioinformatics processing and analysis required for their interpretation. |  13 |
| 12 | For a genomic region of interest (e.g. the neighborhood of a particular gene), use a genome browser to view nearby genes, transcription factor binding regions, epigenetic information, etc.| 14 |
| 13 | Given an RNA-Seq data file, find the set of significantly differentially expressed genes and use online tools to interpret gene lists and annotate potential gene functions. | 15, 16 |
| 14 | Perform a GO analysis to identify the pathways relevant to a set of genes (e.g. identified by transcriptomic study or a proteomic experiment). | 16 |
| 15 | Use the KEGG pathway database to look up interaction pathways. | 17 |
| 16 | Use graph theory to represent biological data networks. | 17, 18 |
| 17 | Understand the challenges in integrating and interpreting large heterogenous high throughput data sets into their functional context. | 19 |
| 18 | Have an appreciation for the social impacts and ethical implications of how genomic sequence information is used in our society | 20 |

 -->
