---
layout: page
title: Schedule
menu: true
order: 2
---


For the Spring 2023 quarter we will meet twice a week on Wednesday and Friday at 12 - 3 pm in YORK 3050 ([map](https://goo.gl/maps/V7xUaJyUDxiGzxyBA){:target="_blank"}). Clicking on the topics below will take you to supporting class content including video lectures, hands-on "lab session" sheets, walk-through screencasts, required reading material and homework assignments.

<br>

| \# | Date         | Topics for Spring 2023                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
| :-: | :-----------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| 0  | - | [**Getting Oriented**](#0) <br> Course introduction, Learning goals & expectations, Meet the instructional team. Setup your computer with required software.                                                                                                                                               |
| 1  | Wed 04/05/23 | [**Welcome to bioinformatics**](#1) <br> Biology is an information science. History of Bioinformatics. Types of data. Application areas and introduction to upcoming course segments. Hands on with major Bioinformatics databases and key online NCBI and EBI resources.                                                                                                                                                |
| 2  | Fri 04/07/23 | [**Sequence alignment fundamentals, algorithms and applications**](#2) <br> Homology. Sequence similarity. Local and global alignment: classic Needleman-Wunsch, Smith-Waterman, and BLAST heuristic approaches. Hands on with dot plots, Needleman-Wunsch, and BLAST algorithms highlighting their utility and limitations.                                                                                                                                                 |
| 3*  | Wed 04/12/23 | **Project:** [**Find a gene project assignment**](#21) <br> (Part 1) Principles of database searching, due in 2 weeks. (Part 2) Sequence analysis, structure analysis and general data analysis with R due at the end of the quarter.                                                                                                                                                                                                 |
| 3  | Wed 04/12/23 | [**Advanced sequence alignment and database searching**](#3) <br> Detecting remote sequence similarity. Database searching beyond BLAST. Substitution matrices. Using PSI-BLAST, Profiles, and HMMs. Protein structure comparisons as a gold standard.                                                                                                                                                                             |
| 4  | Fri 04/14/23 | [**Bioinformatics data analysis with R**](#4) <br> Why do we use R for bioinformatics? R language basics and the RStudio IDE. Major R data structures and functions. Using R interactively from the RStudio console. Introducing Rmarkdown documents.                                                                                              |
| 5  | Wed 04/19/23  | [**Data exploration and visualization in R**](#5) <br> The exploratory data analysis mindset. Data visualization best practices. Simple base graphics (including scatterplots, histograms, bar graphs, dot chats, boxplots, and heatmaps). Building more complex charts with ggplot.                                                                                                                                                                                                                                                    |
| 6  | Fri 04/21/23 | [**Why, when and how of writing your own R functions**](#6) <br> The basics of writing your own functions that promote code robustness, reduce duplication, and facilitate code re-use. Extending functionality and utility with R packages from CRAN and BioConductor. Working with Bio3D for molecular data.                                                                                                                                                                                                                                                                                       |
| 7  | Wed 04/26/23 | [**Introduction to machine learning for bioinformatics**](#7) <br> Unsupervised learning. K-means clustering. Hierarchical clustering. Heatmap representations. Dimensionality reduction. Principal Component Analysis (PCA).                                                                                                                                                                                                                                            |
| 8  | Fri 04/28/23 | [**Unsupervised learning mini-project**](#8) <br> Longer hands-on session with unsupervised learning analysis of cancer cells. Practical considerations and best practices for the analysis and visualization of high dimensional datasets.                                                                                                                                                                                                            |
| 9 | Wed 05/03/23  | [**Structural bioinformatics**](#9) <br> Protein structure function relationships. Protein structure and visualization resources. Modeling energy as a function of structure. Homology modeling. AlphaFold. Predicting functional dynamics. Inferring protein function from structure.                                                                                                                                                                                                                                             |
| 10| Fri 05/05/23 | [**Halloween candy mini-project**](#10) <br> A fun and topical mini-project with unsupervised learning analysis of halloween_candy. Practical considerations and best practices for the exploratory analysis and visualization of high dimensional datasets.                                                                                                                                                                              |
| 11| Wed 05/10/23 | [**Genome informatics and high throughput sequencing**](#11) <br> Searching genes and gene functions. Genome databases. Variation in the genome. High-throughput sequencing technologies, biological applications, and bioinformatics analysis methods. The Galaxy platform along with resources from the EBI & UCSC.                                                                                                                                                                              |
| 12  | Fri 05/12/23 | [**Transcriptomics and the analysis of RNA-Seq data**](#12) <br> RNA-Seq aligners. Differential expression tests. RNA-Seq statistics. Counts, FPKMs, and avoiding P-value misuse. Hands-on analysis of RNA-Seq data with R. Gene functional annotation. Functional databases KEGG, InterPro, GO ontologies, and functional enrichment.                                                                                                                                                                                                                                               |
| 13  |  Wed 05/17/23 | [**RNA-Seq mini-project**](#13) <br> Differential expression analysis project with DESeq2 followed by gene enrichment and functional annotation with KEGG, InterPro, and GO ontologies.                                                                                                                                                                                                                                                                                  |
| 14  | Fri 05/19/23  | [**Hands-on with Git and GitHub**](#14) <br> Why you should use a version control system. How to perform common operations with Git. Creating and working with your own GitHub repos and navigating and using those of others.                                                                                                                         |
| 15  | Wed 05/24/23  | [**Essential UNIX for bioinformatics**](#15) <br> Bioinformatics on the command line. Understanding processes. File system structure. Connecting to remote servers. Redirection, streams, and pipes. Workflows for batch processing. Launching and using AWS EC2 instances (or virtual machines).                                                                                                                                                                                                                                              |
| 16  | Fri 05/26/23  | [**Analyzing sequencing data in the cloud mini-project**](#16) <br> A mini-project using AWS EC2 to query, download, decompress and analyze large data sets from the Sequence Read Archive. Practical considerations and best practices for installing bioinformatics software on Linux, transfering large data sets, and performing analysis either locally or on AWS.                                                                                                                                                                                                                                      |
| 17  | Wed 05/31/23  | [**Vaccination rate mini-project**](#17) <br> A topical mini-project using ggplot and dplyr with the latest state wide COVID-19 vaccination data. Practical considerations and best practices for the exploratory analysis.                                                                                                                                                                                                                                      |
| 18  | Fri 06/02/23  | [**Mutational signatures of cancer mini-project**](#18) <br> Cancer genomics resources and bioinformatics tools for investigating the molecular basis of cancer. Large scale cancer sequencing projects. The Cancer Genome Atlas project, the cBioPortal platform, and the COSMIC database. *De novo* extraction and assignment of mutational signatures.                                                                                                                   |
| 19  | Wed 06/07/23  | [**Pertussis resurgence mini-project**](#19) <br> A topical mini-project using web-scraping, JSON based APIs, and advanced dplyr and ggplot to investigate brand new datasets associated with pertussis cases and longitudinal RNA-Seq on the immune response to vaccination. <br>**Project: Find a gene assignment due\!**                                                                                                                                   |
| 20 | Fri 06/09/23 | [**Portfolio building and discussion of bioinformatics in industry**](#20) <br>Course summary and review. Making a public facing GitHub pages portfolio of your bioinformatics work. <br>Livestream interview with leading bioinformatics and genomics scientists from industry. <br>                                                                                                                                                                                                               |



# Class material

---



<a name="0"></a>
## 0: Getting oriented

**Topics:**  
Course introduction. Learning goals & expectations. Meet the instructional team. Seting up your computer with required software.

**Goals:**  
- Understand course scope, expectations, logistics and ethics code.
- Complete the pre-course questionnaire.
- Setup your computer for this course.


**Videos:**
- 0.1 - [Welcome to BIMM143 (course introduction and overview)](https://youtu.be/hXd58ESRf9w){:target="_blank"}  
- 0.2 - [Website overview (finding course content and installing software)](http://youtu.be/DjUZaPHxObk){:target="_blank"} 
> **Note:** format and links on the current website may differ from those in video.


**Supporting material:**  
- Handout: [Class Syllabus]({{ site.baseurl}}{{site.data.authors.corelinks.syllabus}}){:.no-push-state}{:target="_blank"}
- Pre-course [Questionnaire]({{ site.data.authors.corelinks.questionnaire }}){:target="_blank"}
- Computer [Setup Instructions]({{ site.baseurl }}/setup/)
- Sign up for our [Piazza class Q&A site]({{ site.data.authors.corelinks.piazza }}){:target="_blank"}
- View the class [**GradeBook**]({{ site.data.authors.corelinks.gradebook }}){:target="_blank"}


---

<a name="1"></a>
## 1: Welcome to bioinformatics

**Topics:**  
Biology is an information science. History of Bioinformatics. Types of data. Application areas and introduction to upcoming course segments. Introduction to NCBI & EBI resources for the molecular domain of bioinformatics. Hands-on session using NCBI-BLAST, Entrez, GENE, UniProt, Muscle, and PDB bioinformatics tools and databases.

**Goals:**
- Understand the increasing necessity for computation in modern life sciences research.  
- Get introduced to how bioinformatics is practiced. 
- Be able to query, search, compare and contrast the data contained in major bioinformatics databases (GenBank, GENE, UniProt, PFAM, OMIM, PDB) and describe how these databases intersect. 
- The goals of the hands-on session is to introduce a range of core bioinformatics databases and associated online services while actively investigating the molecular basis of several common human diseases.  


**Videos:**
- 1.1 - [Introduction to bioinformatics (what, where and why of bioinformatics)](https://youtu.be/FNXMFXZR6HU){:.no-push-state}{:target="_blank"},  
- 1.2 - [Major bioinformatics resource providers (NCBI and EBI)](https://youtu.be/o_IO7dDUZPM){:.no-push-state}{:target="_blank"},  
- 1.3 - [A quick tour of major NCBI and EBI resources (GENE, UniProt, GO, OMIM, PDB, PFAM)](https://youtu.be/afqB87ZSkXw){:.no-push-state}{:target="_blank"}.    

**Hands-on Lab:**
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/class1/lab1_sp23.pdf){:target="_blank"}{:.no-push-state}
> Feedback: [Muddy Point Assessment](https://forms.gle/L2Lpzm4aWUvwvA7n8){:.no-push-state}{:target="_blank"}

**Supporting Material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class1/lecture1-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class1/lecture1-BIMM143-small.pdf){:.no-push-state}{:target="_blank"}  
- Handout: [Major Bioinformatics Databases]({{ site.baseurl }}/class-material/class1/Major_Databases_BIMM143.pdf){:.no-push-state}{:target="_blank"}
- Screencast Lab: [Video walk-through](https://youtu.be/ps0BkDZaiwo){:target="_blank"}{:.no-push-state}
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"}

**Homework:**  
- [Quiz](https://forms.gle/iEBpDJz4qpiNsUit9){:.no-push-state}{:target="_blank"},  
- Submit your completed lab report (i.e. filled in PDF form) to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}
- Readings: 
  - PDF1: [What is bioinformatics? An introduction and overview]({{ site.baseurl }}/class-material/class1/bioinformatics_review.pdf){:.no-push-state}
  - PDF2: [Advancements and Challenges in Computational Biology]({{ site.baseurl }}/class-material/class1/bioinformatics_challenges_2015.pdf){:.no-push-state}

---

<a name="2"></a>
## 2: Sequence alignment fundamentals, algorithms and applications

**Topics:**  
Sequence Alignment and Database Searching: Homology, Sequence similarity, Local and global alignment, Heuristic approaches, Database searching with BLAST, E-values and evaluating alignment scores and statistics.

**Goals:**
- Be able to describe how dynamic programming works for pairwise sequence alignment.  
- Appreciate the differences between global and local alignment along with their major application areas.  
- Understand how aligning novel sequences with previously characterized genes or proteins provides important insights into their common attributes and evolutionary origins.  
- The goals of the hands-on session are to explore the principles underlying the computational tools that can be used to compute and evaluate sequence alignments.  


**Videos:**  
- 2.1 - [Alignment fundamentals](https://www.youtube.com/watch?v=SU22pycFrDk){:.no-push-state}{:target="_blank"}
- 2.2 - [Dot plots](http://youtu.be/nnPNwlzX2qg){:.no-push-state}{:target="_blank"}
- 2.3 - [Dynamic programing, global alignment](https://youtu.be/LkwgI2mHbik){:.no-push-state}{:target="_blank"}
- 2.4 - [Dynamic programing, local alignment and BLAST basics](https://youtu.be/uhD77hnxM2g){:.no-push-state}{:target="_blank"}

**Hands-on Lab:**
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/class2/lab2_sp23.pdf){:target="_blank"}{:.no-push-state}
> Feedback: [Muddy Point Assessment](https://forms.gle/JwKrRCmdcYFjsmP4A){:.no-push-state}{:target="_blank"}

**Supporting Material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class2/lecture2-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class2/lecture2-BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- Dot Plot App Mirrors: [app-1](https://barryus.shinyapps.io/dotplot/){:.no-push-state}{:target="_blank"},  [app-2](https://bioboot.shinyapps.io/dotplot/){:.no-push-state}{:target="_blank"}
- Screencast Lab: [Video walk-through](https://youtu.be/U4u8jJqA2PI){:target="_blank"}{:.no-push-state}
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"} 

**Homework:**   
- [Quiz](https://forms.gle/MgQbfcN3K1tAVs8L8){:.no-push-state}{:target="_blank"}
- Submit your completed lab report (i.e. filled in PDF form) answering questions **Q1 to Q6** to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}  
- Complete the following [Alignment Problem]({{ site.baseurl }}/class-material/class2/wk2_homework.pdf){:.no-push-state}{:target="_blank"} and submit it to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}  
- For next week please [install R and RStudio]({{ site.baseurl }}/setup/){:.no-push-state}{:target="_blank"}
- DataCamp: Sign-up to our **BIMM143_DiazGay group/organization** via the link on Piazza or in your UCSD email. We will use this from next week onward. However, feel free to get started with your first course [**Introduction to R!**](https://learn.datacamp.com/courses/free-introduction-to-r){:.no-push-state}{:target="_blank"} 

**Readings:**   
- Readings: PDF1: [What is dynamic programming?]({{ site.baseurl }}/class-material/class2/Dynamic_programming_primer.pdf){:.no-push-state}
- Readings: PDF2 [Fundamentals of database searching]({{ site.baseurl }}/class-material/class2/Fundamentals.pdf){:.no-push-state}



---
<a name="21"></a>



## 3*: (**Project**) Find a gene assignment
The [**find-a-gene project**]({{ site.baseurl }}/class-material/project/Find_A_Gene_Project.pdf){:.no-push-state} is a required assignment for BIMM 143. The objective with this assignment is for you to demonstrate your grasp of database searching, sequence analysis, structure analysis and the R environment that we have covered to date in class.

You may wish to consult the scoring rubric at the end of the above linked project description and the [**example report**]({{ site.baseurl }}/class-material/project/Find_A_Gene_Project_Example.pdf){:.no-push-state} for format and content guidance.

- Your responses to questions Q1-Q4 are due **in three weeks** time (Tuesday, May 2 at 12pm).

- The complete assignment, including responses to all questions, is due **Tuesday of week 10** (June 6) at 12pm.

- In both instances your PDF format report should be submitted to GradeScope. Late responses will not be accepted under any circumstances.

**Videos:**  
- 3.1 - [Project introduction](https://youtu.be/t0EfxbEpvW4){:.no-push-state}{:target="_blank"}
> **Note:** due dates may differ from those in video.    



---
<a name="3"></a>

## 3: Advanced sequence alignment and database searching

**Topics**:  
Detecting remote sequence similarity. Substitution matrices. Database searching beyond BLAST with PSI-BLAST. Profiles and HMMs. Protein structure comparisons. Beginning with command line based database searches.

**Goals:**
- Be able to calculate the alignment score between protein (or nucleotide) sequences using a provided scoring matrix such as BLOSUM62.  
- Understand the limits of homology detection with tools such as BLAST.  
- Know how to derive a PROSITE style regular expression for aligned motifs.  
- Be able to calculate a PSSM profile and for aligned sequences and subsequently score new sequences using a PSSM.  
- Be able to perform PSI-BLAST, HMMER and protein structure based database searches and interpret the results in terms of the biological significance of an e-value.  
- Be familiar with the concepts of True Positives, False Positives, Sensitivity and Specificity.  

**Hands-on Lab:**   
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/class3/lab3_sp23.pdf){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy Point Assessment](https://forms.gle/NFriqivMdgNUzMD96){:.no-push-state}{:target="_blank"}

**Supporting Material:**
 - Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class3/lecture3-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class3/lecture3-BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- Bonus: [Alignment App](https://marcos-diazg.github.io/BIMM143_SP23/class-material/class3/nw/){:.no-push-state}{:target="_blank"}

**Homework:**   
- [Questions](https://docs.google.com/document/d/1ruSvmoUzmOX77t81LdIPy7xjxav0OcPWRHHCKslspGg/copy){:.no-push-state}{:target="_blank"} click and select "make a copy" then follow instructions
- Submit your completed lab report (i.e. filled in PDF form) answering questions **Q1 to Q7** to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}  
- **DataCamp**: Sign-up to our [**BIMM143_DiazGay** group/organization](https://app.datacamp.com/groups/bimm143_diazgay/dashboard){:.no-push-state}{:target="_blank"} via the link in your UCSD email and **start** (you do not have to finish yet) [**Introduction to R!**](https://learn.datacamp.com/courses/free-introduction-to-r){:.no-push-state}{:target="_blank"} (we will complete this next week).  
- [RStudio and R download and setup]({{ site.baseurl }}/setup/)



---
<a name="4"></a>
## 4: Bioinformatics data analysis with R


**Topics:**  
Why do we use R for bioinformatics? R language basics and the RStudio IDE. Major R data structures and functions. Using R interactively from the RStudio console.

**Goals:**
- Understand why we use R for bioinformatics.
- Familiarity with Râ€™s basic syntax.
- Familiarity with major R data structures (vectors, data.frames and lists).
- Understand the basics of using functions (arguments, vectorizion and re-cycling).


**Videos:**
- 4.1 [Why R and RStudio](https://youtu.be/Asm2PHOZAcE){:.no-push-state}{:target="_blank"}
- 4.2 [Major R data structures, data types, and using functions](http://youtu.be/3LOTxeQEHSM){:.no-push-state}{:target="_blank"}
- 4.3 [Working with DataCamp](https://www.youtube.com/watch?v=q_GUl2YRoVk){:.no-push-state}{:target="_blank"}
> **Note:** Use your UCSD email invite to sign up and visit our class group/organization

**Hands-on Lab:**
- Lab: [Hands-on section]({{ site.baseurl }}/class-material/class4/lab4_sp23/){:.no-push-state} 
> Feedback: [Muddy point assessment](https://forms.gle/Vo1cn1j2NUHzjPH37){:.no-push-state}{:target="_blank"}

**Supporting Material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class4/lecture4-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class4/lecture4-BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- Cheat Sheet: [Base R overview]({{ site.baseurl }}/class-material/class4/base-r.pdf){:target="_blank"}{:.no-push-state}
- Screencast Lab: [Video walk-through](https://youtu.be/QQ7KPg2NnpY?t=1098){:target="_blank"}{:.no-push-state} focusing on introducing R data structures and core syntax
- Optional extension: [Advanced conservation analysis of globins with R]({{ site.baseurl }}/class-material/class4/globin_analysis){:.no-push-state} this demonstrates where we are going on our R learning journey. You should be able to do analysis like this on your own at the end of the course.  
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"}

**Homework:**
- [Quiz](https://forms.gle/zLducArgkPY8ij5Z8){:.no-push-state}{:target="_blank"}
- Submit your lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}  
- **DataCamp**: Sign-up to our **BIMM143_DiazGay group/organization** via the link in your UCSD email and **complete** [**Introduction to R!**](https://learn.datacamp.com/courses/free-introduction-to-r){:.no-push-state}{:target="_blank"} (~4hrs)

**[OPTIONAL] Extra Credit:**
- Extra credit lab: [Introduction to data in R]({{ site.baseurl }}/class-material/class4/ec4_sp23.pdf){:.no-push-state}{:target="_blank"}



---
<a name="5"></a>
## 5: Data exploration and visualization in R

**Topics:**  
The exploratory data analysis mindset. Data visualization best practices. Simple base graphics (including scatterplots, histograms, bar graphs, dot chats, boxplots, and heatmaps). Building more complex charts with ggplot.


**Goals:**
- Appreciate the major elements of exploratory data analysis and why it is important to visualize data.  
- Be conversant with data visualization best practices and understand how good visualizations optimize for the human visual system.  
- Be able to generate informative graphical displays including scatterplots, histograms, bar graphs, boxplots, dendrograms and heatmaps and thereby gain exposure to the extensive graphical capabilities of R.  
- Appreciate that you can build even more complex charts with ggplot and additional R packages.  
- Be able to write and (re)use basic R scripts to aid with reproducibility.  



**Videos:**
- 5.1 - [Why visualize data?](https://youtu.be/R_b7g5sGzwY){:.no-push-state}{:target="_blank"}  
- 5.2 - [Data visualization best practices](https://youtu.be/0WPeOxrboug){:.no-push-state}{:target="_blank"}
- 5.3 - [Introduction to ggplot](https://youtu.be/qtfJa8muH9E){:.no-push-state}{:target="_blank"}
<!-- - 5.4 - Optional: [The **g**rammar of **g**raphics - the **gg** in ggplot](https://vimeo.com/332290655){:.no-push-state}{:target="_blank"},    -->

**Hands-on Lab:**
- Lab: [Hands-on worksheet]({{ site.baseurl }}/class-material/class5/lab5_sp23.html){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy point assessment](https://forms.gle/5uykvYTHTxvWjVfD9){:.no-push-state}{:target="_blank"}

**Supporting Material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class5/lecture5-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class5/lecture5-BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- Screencast Lab: [Video walk-through](https://youtu.be/p9ap6X1V3mo){:target="_blank"}{:.no-push-state}
- Side-Note: [Convincing with graphics](https://xkcd.com/833/){:.no-push-state}
- Check-out the [Data-to-Viz](https://www.data-to-viz.com/){:.no-push-state}{:target="_blank"} website and the [ggplot cheat sheat](https://posit.co/wp-content/uploads/2022/10/data-visualization-1.pdf){:.no-push-state}{:target="_blank"}
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"}
 
**Homework:**   
- [Quiz](https://forms.gle/aDujhaRG3rtHUEoJ8){:.no-push-state}{:target="_blank"} 
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}
- DataCamp: [Introduction to data visualization with ggplot2](https://learn.datacamp.com/courses/introduction-to-data-visualization-with-ggplot2){:.no-push-state}{:target="_blank"} (~4hrs)   


 
---
<a name="6"></a>
## 6: Why, when and how of writing your own R functions

**Topics:**  
The why, when and how of writing your own R functions with worked examples. Further extending functionality and utility with R packages. Obtaining R packages from CRAN and Bioconductor. Working with Bio3D for molecular data. Managing genome-scale data with Bioconductor.

**Goals:**
- Understand the structure and syntax of R functions and how to view the code of any R function.
- Be able to follow a step by step process of going from a working code snippet to a more robust function that reduces duplication and facilitate code re-use.
- Be able to find and install R packages from CRAN and bioconductor.
- Understand how to find and use package vignettes, demos, documentation, tutorials and source code repository where available.

**Videos:**  
- 6.1 - [Writing your own functions (why, when and how)](https://youtu.be/NzIHZyywsAo){:.no-push-state}{:target="_blank"}
- 6.2 - [Introduction to CRAN & BioConductor](https://youtu.be/qAvJ92qCGqE){:.no-push-state}{:target="_blank"}
- 6.3 - [Quick introduction to RMarkdown](https://www.youtube.com/watch?v=O7GMs9V3HQk){:.no-push-state}{:target="_blank"}
- 6.4 - Optional longer video: [Getting started with RMarkdown](https://rstudio.com/resources/webinars/getting-started-with-r-markdown/){:.no-push-state}{:target="_blank"}  

**Hands-on Lab:**
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/class6/lab06_sp23.pdf){:.no-push-state}{:target="_blank"}
- Lab supplement: [Hands-on section supplemental information]({{ site.baseurl }}/class-material/class6/lecture6-BIMM143_F20.pdf){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy point assessment](https://forms.gle/4Tnks8NZykGaABMaA){:.no-push-state}

**Supporting material:**  
- Lecture Slides: [Pt1. Large PDF]({{ site.baseurl }}/class-material/class6/lecture06-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Pt2. Large PDF]({{ site.baseurl }}/class-material/class6/lecture07-BIMM143-large.pdf){:.no-push-state}{:target="_blank"}
- Screencast Lab: [video walk-through](https://youtu.be/Zr9BVGGhaNc){:target="_blank"}{:.no-push-state}  
- Bonus: [Introductory tutorial on R packages](https://www.datacamp.com/community/tutorials/r-packages-guide){:.no-push-state}{:target="_blank"}
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"} 

 
**Homework:**   
- [Quiz](https://forms.gle/7THKTrvjA2ChJrX58){:.no-push-state}{:target="_blank"}
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}
- **Write a function**: See **Q6** of the [hands-on lab supplement above]({{ site.baseurl }}/class-material/class6/lecture6-BIMM143_F20.pdf){:.no-push-state}{:target="_blank"}. This entails turning a supplied code snippet into a more robust and re-usable function that will take any of the three listed input proteins and plot the effect of drug binding. Note assessment rubric and submission instructions within document
- DataCamp: Please **complete** the [Introduction to ggplot2](https://learn.datacamp.com/courses/introduction-to-data-visualization-with-ggplot2){:.no-push-state}{:target="_blank"} course


**Other**:  
- Flat files for practicing importing with read.table: [test1.txt]({{ site.baseurl }}/class-material/class6/test1.txt){:.no-push-state} / [test2.txt]({{ site.baseurl }}/class-material/class6/test2.txt){:.no-push-state} / [test3.txt]({{ site.baseurl }}/class-material/class6/test3.txt){:.no-push-state}



---
<a name="7"></a>
## 7: Introduction to machine learning for bioinformatics

**Topics:**  
Unsupervised learning, supervised learning, and reinforcement learning. Focus on unsupervised learning. K-means clustering. Hierarchical clustering. Dimensionality reduction, visualization, and analysis. Principal Component Analysis (PCA). Practical considerations and best practices for the analysis of high dimensional datasets.


**Goals:**  
- Understand the major differences between unsupervised and supervised learning.  
- Be able to create k-means and hierarchical cluster models in R.
- Be able to describe how the k-means and bottom-up hierarchical cluster algorithms work.  
- Know how to visualize and integrate clustering results and select good cluster models.  
- Be able to describe in general terms how PCA works and its major objectives.  
- Be able to apply PCA to high dimensional datasets and visualize and integrate PCA results (e.g., identify outliers, find structure in features and aid in complex dataset visualization).



**Videos:**
- 7.1 - [Introduction to unsupervised learning and K-means clustering](http://youtu.be/k2QPXRtTons){:.no-push-state}{:target="_blank"}
- 7.2 - [Hierarchical clustering](https://youtu.be/lU8S_Do3OPQ){:.no-push-state}{:target="_blank"}
- 7.3 - [Principal component analysis (PCA) Pt.1](http://youtu.be/Olm0gW6OXIg){:.no-push-state}{:target="_blank"} 

**Hands-on Lab:**
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/class7/lab07_sp23.html){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy point assessment](https://forms.gle/oPbcfpZofEx28Wqz6){:.no-push-state}.  

**Supporting material:**  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class7/lecture08-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class7/lecture08-BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- WebApp: [Introduction to PCA]({{ site.baseurl }}/class-material/class7/pca/){:.no-push-state}{:target="_blank"}
- Data files: [UK_foods.csv]({{ site.baseurl }}/class-material/class7/UK_foods.csv){:.no-push-state} / [expression.csv]({{ site.baseurl }}/class-material/class7/expression.csv){:.no-push-state} 
- Screencast Lab: [video walk-through](https://youtu.be/SERsmSnwpnk){:target="_blank"}{:.no-push-state}
- Bonus: [StackExchange discussion on PCA](https://stats.stackexchange.com/questions/2691/making-sense-of-principal-component-analysis-eigenvectors-eigenvalues?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa){:.no-push-state}{:target="_blank"}  
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"}

**Homework:**   
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}
- DataCamp: [Introduction to the Tidyverse](https://learn.datacamp.com/courses/introduction-to-the-tidyverse){:.no-push-state}{:target="_blank"} (~4hrs)



---
<a name="8"></a>
## 8: Unsupervised learning mini-project

**Topics:**  
Hands-on project session with unsupervised learning analysis of cancer cells. Practical considerations and best practices for the analysis and visualization of high dimensional datasets.  

**Goals:**
- Be able to import and prepare data for unsupervised learning analysis.
- Be able to apply and test combinations of PCA, k-means, and hierarchical clustering to high dimensional datasets and critically review results.

**Hands-on Lab:**
- Lab: [Mini-Project]({{ site.baseurl }}/class-material/class8/lab08_sp23.html){:.no-push-state}{:target="_blank"} 
> Feedback: [Muddy point assessment](https://forms.gle/j9qhXUNoqBWfyCzPA){:.no-push-state} 

**Supporting material:**  
- Data file: [WisconsinCancer.csv]({{ site.baseurl }}/class-material/class8/WisconsinCancer.csv){:.no-push-state} / [new_samples.csv]({{ site.baseurl }}/class-material/class8/new_samples.csv){:.no-push-state}
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class8/lecture09-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class8/lecture09-BIMM143-small.pdf){:.no-push-state}{:target="_blank"}

**Homework:**   
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}
- DataCamp: **Complete** [Introduction to the Tidyverse](https://learn.datacamp.com/courses/introduction-to-the-tidyverse){:.no-push-state}{:target="_blank"} (~4hrs)



---
<a name="9"></a> 
## 9: Structural bioinformatics

**Topics:**  
Protein structure function relationships. Protein structure and visualization resources. Modeling energy as a function of structure. Homology modeling. AlphaFold. Predicting functional dynamics. Inferring protein function from structure.

**Goals:**
- View and interpret the structural models in the PDB.
- Understand the classic `Sequence > Structure > Function` via energetics and dynamics paradigm.
- Be able to use VMD for biomolecular visualization and analysis.
- Appreciate how AlphaFold has advanced structural bioinformatics.
- Be able to use the Bio3D package for exploratory analysis of protein  sequence-structure-function-dynamics relationships.    

**Videos:**  
- 9.1 - [Introduction to structural bioinformatics](https://www.youtube.com/watch?v=lLLXkdTl8aQ){:.no-push-state}{:target="_blank"}
- 9.2 - [Visualization, interpretation and modeling of protein structure](https://www.youtube.com/watch?v=8GO9kUOPfMQ){:.no-push-state}{:target="_blank"}
- 9.3 - [The story of AlphaFold](https://www.youtube.com/watch?v=gg7WjuFs8F4){:.no-push-state}{:target="_blank"}

**Hands-on Lab:**
- Lab: [Hands-on section]({{ site.baseurl }}/class-material/class9/pdb_pca_labclass.html){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy point assessment](https://forms.gle/9c5JogJrcor5s1qH7){:.no-push-state}


**Supporting material:**  
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class9/lecture11-BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class9/lecture11-BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- Software links: [VMD download](http://www.ks.uiuc.edu/Development/Download/download.cgi){:.no-push-state}{:target="_blank"}

<!-- / [MUSCLE download](https://www.drive5.com/muscle/downloads.htm){:.no-push-state}{:target="_blank"} -->
<!-- - Alternate Windows install and setup cmd:<br>`curl -o "muscle.exe" "https://www.drive5.com/muscle/downloads3.8.31/muscle3.8.31_i86win32.exe"`   
- Alternative Mac install and setup cmd:<br>`sudo curl -o "/usr/local/bin/muscle" "http://thegrantlab.org/misc/muscle"; sudo chmod +x /usr/local/bin/muscle`  
- Alternative M1 Mac install and setup cmd:<br>`sudo curl -o "/usr/local/bin/muscle" "http://thegrantlab.org/misc/m1/muscle"; sudo chmod +x /usr/local/bin/muscle`   -->


**Homework:**
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}



---
<a name="10"></a>
## 10: Halloween candy mini-project

**Topics:**  
A fun and topical mini-project with unsupervised learning analysis of halloween_candy. Practical considerations and best practices for the analysis and visualization of high dimensional datasets.

**Hands-on Lab:**
- Lab: [Mini-Project]({{ site.baseurl }}/class-material/class10/Halloween_candy.html){:.no-push-state}{:target="_blank"}

**Homework:**
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}


---
<a name="11"></a>
## 11: Genome informatics

**Topics:**  
Genome sequencing technologies past, present and future (Sanger, Shotgun, PacBio, Illumina, toward the $500 human genome). Biological applications of sequencing. Variation in the genome. RNA-Sequencing for gene expression analysis. Major genomic databases, tools, and visualization resources from the EBI & UCSC. The Galaxy platform for quality control and analysis. Sample Galaxy RNA-Seq workflow with FastQC and Bowtie2.

**Goals:**
- Appreciate and describe in general terms the rapid advances in sequencing technologies and the new areas of investigation that these advances have made accessible.  
- Understand the process by which genomes are currently sequenced and the bioinformatics processing and analysis required for their interpretation.  
- For a genomic region of interest (e.g. the neighborhood of a particular SNP), use a genome browser to view nearby genes, transcription factor binding regions, epigenetic information, etc.  
- Be able to use the Galaxy platform for basic RNA-Seq analysis from raw reads to expression value determination.  
- Understand the FASTQ file format and the information it holds.  
- Understand the SAM/BAM file format and the information it holds.  

**Videos:**  
- 11.1 - [Introduction to genomics](http://youtu.be/QWOYBFfsKIE){:.no-push-state}{:target="_blank"}
- 11.2 - [Sequencing methods](https://www.ibiology.org/techniques/dna-sequencing/){:.no-push-state}{:target="_blank"} from Jonathan Weissman (UCSF)
- 11.3 - [The basics of RNASeq work-flows](http://youtu.be/PCn9GiD4cUI){:.no-push-state}{:target="_blank"}
- 11.4 - Optional: [Lessons from the Human Genome Project](https://www.youtube.com/watch?v=qOW5e4BgEa4){:.no-push-state}{:target="_blank"}

**Hands-on Lab:**
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/class11/wk8_genomics_lab.pdf){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy point assessment](https://forms.gle/buzVikzeQZvdvU4S8){:.no-push-state}{:target="_blank"}

**Supporting material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class11/lecture13_BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class11/lecture13_BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- Screencast Lab: [video walk-through](https://youtu.be/jfAlMYAJNw8){:target="_blank"}{:.no-push-state} 
- [Galaxy Server](https://usegalaxy.org){:target="_blank"}{:.no-push-state}, create a free account for section 3 of the lab
- RNA-Seq data files: [HG00109_1.fastq]({{ site.baseurl }}/class-material/class11/HG00109_1.fastq){:.no-push-state} / [HG00109_2.fastq]({{ site.baseurl }}/class-material/class11/HG00109_2.fastq){:.no-push-state} / [genes.chr17.gtf]({{ site.baseurl }}/class-material/class11/genes.chr17.gtf){:.no-push-state} / [Expression genotype results]({{ site.baseurl }}/class-material/class11/rs8067378_ENSG00000172057.6.txt){:.no-push-state}
- [SAM/BAM file format description]({{ site.baseurl }}//class-material/class11/sam_format/){:.no-push-state}{:target="_blank"}
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"}

**Homework:**
- Submit your RMarkdown/Quarto generated PDF lab report answering questions **Q1 to Q12** to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}

**[OPTIONAL] Extra Credit:**
- **Population analysis**: Submit to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"} your RMarkdown/Quarto generated PDF with working code, output and narrative text answering **Q13 and Q14** in this week's [hands-on section worksheet]({{ site.baseurl }}/class-material/class11/wk8_genomics_lab.pdf){:.no-push-state}{:target="_blank"}



---
<a name="12"></a>
## 12: Transcriptomics and the analysis of RNA-Seq data

**Topics:**  
Analysis of RNA-Seq data with R. Differential expression tests. RNA-Seq statistics. Counts and FPKMs. Normalizing for sequencing depth. DESeq2 analysis. Gene finding and functional annotation from high throughput sequencing data. Functional databases KEGG, InterPro, GO ontologies, and functional enrichment.  


**Goals:**  
- Given an RNA-Seq dataset, find the set of significantly differentially expressed genes and their annotations.  
- Gain competency with data import, processing and analysis with DESeq2 and other Bioconductor packages.  
- Understand the structure of count data and metadata required for running analysis.  
- Be able to extract, explore, visualize and export results.  
- Perform a GO analysis to identify the pathways relevant to a set of genes (e.g. identified by transcriptomic study or a proteomic experiment). Use both Bioconductor packages and online tools to interpret gene lists and annotate potential gene functions.

**Videos:**
- 12.1 - [Differential expression analysis of RNA-Seq data](http://youtu.be/vn5Fy2VVWBw){:.no-push-state}{:target="_blank"}
- 12.2 - [Differential expression tests and pathway analysis](http://youtu.be/EIMK4TSnQLE){:.no-push-state}{:target="_blank"}
- 12.3 - [Installing Bioconductor and DESeq2](http://youtu.be/aekKXg0U1Rg){:.no-push-state}{:target="_blank"} 

**Hands-on Lab:**
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/class12/lab12_sp23.html){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy point assessment](https://forms.gle/C2F2Q84T9d72ixfZ8){:.no-push-state}

**Supporting material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class12/lecture14_BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class12/lecture14_BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- Detailed [Bioconductor setup]({{ site.baseurl }}//class-material/class12/bioconductor_setup/){:.no-push-state}{:target="_blank"} instructions
- Screencast Lab: [video walk-through](https://youtu.be/HnT1YL_KxQk){:target="_blank"}{:.no-push-state}
- Data files: [airway_scaledcounts.csv]({{ site.baseurl }}/class-material/class12/airway_scaledcounts.csv){:.no-push-state} / [airway_metadata.csv]({{ site.baseurl }}/class-material/class12/airway_metadata.csv){:.no-push-state}
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"}

**Readings**:
 - Excellent review article: [Conesa _et al_. A survey of best practices for RNA-seq data analysis. _Genome Biology_ (2016)](http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8){:.no-push-state}
 - An oldey but a goodie: [Soneson _et al_. Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences. _F1000Research_ (2015)](https://f1000research.com/articles/4-1521/v2)
 - Good review article: [Trapnell _et al_. Differential analysis of gene regulation at transcript resolution with RNA-seq. _Nat Biotechnol_ (2013)](https://www.ncbi.nlm.nih.gov/pubmed/23222703){:.no-push-state}
 - Abstract and introduction sections of: [Himes _et al_. RNA-Seq transcriptome profiling identifies _CRISPLD2_ as a glucocorticoid responsive gene that modulates cytokine function in airway smooth muscle cells. _PLoS ONE_ (2014)](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0099625){:.no-push-state}

**Homework:**
- Submit your completed PDF lab report answering questions **Q1 to Q10** to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}



--- 
<a name="13"></a>
## 13: RNA-Seq analysis mini-project

**Topics:**  
Differential expression analysis mini-project with DESeq2 followed by gene enrichment and functional annotation with KEGG, InterPro, and GO ontologies.  

**Hands-on Lab:**
- Lab: [Mini-Project]({{ site.baseurl }}/class-material/class13/lab13_sp23.html){:.no-push-state}{:target="_blank"}
> Feeback [Muddy point assessment](https://forms.gle/dRLYpQYLQfzZfdWo8){:.no-push-state}

**Supporting material:**
- Data files: [GSE37704_featurecounts.csv]({{ site.baseurl }}/class-material/class13/GSE37704_featurecounts.csv){:.no-push-state} / [GSE37704_metadata.csv]({{ site.baseurl }}/class-material/class13/GSE37704_metadata.csv){:.no-push-state}

**Homework:**
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}
- DataCamp: [Intermediate R](https://app.datacamp.com/learn/courses/intermediate-r){:.no-push-state}{:target="_blank"} (~4hrs)


---
<a name="14"></a>
## 14: Hands-on with Git and GitHub

**Topics:**  
Why you should use a version control system. How to perform common operations with Git, currently the most popular version control system. Creating and working with your own GitHub repos and navigating and using those of others. Syncing bioinformatics work to GitHub.     

**Videos:**  
- 14.1 - OPTIONAL: [Git for humans](https://youtu.be/eWxxfttcMts){:.no-push-state}{:target="_blank"}

**Hands-on Lab:**
- Lab: [Hands-on section worksheet]({{ site.baseurl }}/class-material/class14/github_lab.html){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy-Point-Assessment](https://forms.gle/NiBJgDPQAfYyMa9P9){:.no-push-state}

**Supporting material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class14/slide_class16.pdf){:.no-push-state}{:target="_blank"}

**Homework:**   
- Submit your GitHub class repository URL to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}
- DataCamp: **Complete** [Intermediate R](https://app.datacamp.com/learn/courses/intermediate-r){:.no-push-state}{:target="_blank"} (~4hrs)


---
<a name="15"></a>
## 15: Essential UNIX for bioinformatics

**Topics:**  
Bioinformatics on the command line. Why do we use UNIX for bioinformatics? UNIX philosophy. 21 Key commands. Understanding processes. File system structure. Connecting to remote servers. Redirection, streams and pipes. Workflows for batch processing. Organizing computational projects. Going further with your own computer in the cloud. Launching and using AWS EC2 instances (or virtual machines). 


**Goals:**  
- Understand why we use UNIX for bioinformatics.
- Use UNIX command-line tools for file system navigation and text file manipulation.
- Have a familiarity with 21 key UNIX commands that we will use ~90% of the time.
- Be able to connect to remote servers from the command line.
- Use existing programs at the UNIX command line to analyze bioinformatics data.
- Understand IO redirection, streams and pipes.
- Understand best practices for organizing computational projects.

**Videos:**
- 15.1 - [Essential UNIX for bioinformatics I](http://youtu.be/qsDzjQm_Wp0){:.no-push-state}{:target="_blank"}
- 15.2 - [Essential UNIX for bioinformatics II](http://youtu.be/11t1GuZB77M){:.no-push-state}{:target="_blank"}
- 15.3 - [Manipulating files on UNIX machines](http://youtu.be/_jVgj6UutwA){:.no-push-state}{:target="_blank"}
- 15.4 - [UNIX superpowers: using pipes and conecting to remote machines](http://youtu.be/AKLha-_IurU){:.no-push-state}{:target="_blank"}

**Hands-on Lab:**
- Hands-on section worksheet
  * (Part I) [Starting your own computer in the cloud]({{ site.baseurl }}/class-material/class15/aws_01.html){:.no-push-state}{:target="_blank"}
  * (Part II) [Accessing and using your AWS instance]({{ site.baseurl }}/class-material/class15/aws_02.html){:.no-push-state}{:target="_blank"}
- AWS Console URL: [https://awsed.ucsd.edu/](https://awsed.ucsd.edu/){:.no-push-state}{:target="_blank"}
> Feedback: [Muddy point assessment](https://forms.gle/cp2wBCHY7RC97QQz6){:.no-push-state}

**Supporting material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class15/lecture16_BIMM143-large.pdf){:.no-push-state}{:target="_blank"} / [Small PDF]({{ site.baseurl }}/class-material/class15/lecture16_BIMM143-small.pdf){:.no-push-state}{:target="_blank"}
- Screencast Lab: [launching an AWS EC2 instance](http://youtu.be/5-Nxt85UmoA){:.no-push-state}{:target="_blank"}
- Office/Student Hours: [{{ site.data.authors.corelinks.officetime }}]({{ site.data.authors.corelinks.officehours }}){:.no-push-state}{:target="_blank"}

**Homework:**   
- [Questions]({{ site.baseurl }}/class-material/class15/HW8_unix_questions.pdf){:.no-push-state}{:target="_blank"}: complete PDF form with your answers, save, and submit to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}
- _No lab report due this week_
- Bonus: [Introduction to the Unix shell](https://learn.datacamp.com/courses/introduction-to-shell){:.no-push-state}{:target="_blank"} (~4hrs)
 


---
<a name="16"></a>
## 16: Analyzing sequencing data in the cloud mini-project

**Topics:**  
A mini-project using AWS EC2 cloud computing resources to query, download, decompress, and analyze large data sets from the NCBI's Sequence Read Archive (SRA). Practical considerations and best practices for installing bioinformatics software on Linux, transfering large data sets, and performing analysis either locally or on AWS.

**Hands-on Lab:**
- Lab: [Mini-Project]({{ site.baseurl }}/class-material/class16/aws_03.html){:.no-push-state}{:target="_blank"}

**Supporting material:**
- AWS Console URL: [https://awsed.ucsd.edu/](https://awsed.ucsd.edu/){:.no-push-state}{:target="_blank"}

**Homework:**   
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}



---
<a name="17"></a>
## 17: Vaccination rate mini-project

**Topics:**  
A topical mini-project using ggplot and dplyr with the latest state wide COVID-19 vaccination data. Practical considerations and best practices for exploratory data analysis.

**Hands-on Lab:**
- Lab: [Mini-Project]({{ site.baseurl }}/class-material/class17/lab17_sp23.html){:.no-push-state}{:target="_blank"}

**Supporting material:**
- Data files: [Statewide COVID-19 vaccines administered by ZIP code (updated: 2023/02/28)]({{ site.baseurl }}/class-material/class17/covid19vaccinesbyzipcode.csv){:.no-push-state}{:target="_blank"}
- Original source of the COVID-19 vaccination data at the [California Open Data Portal](https://data.ca.gov/dataset/covid-19-vaccine-progress-dashboard-data-by-zip-code){:.no-push-state}{:target="_blank"}

**Homework:**   
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}

<!--
---
<a name="18"></a>
## 18: TBD.  

To be updated.   

 -->


---
<a name="19"></a>
## 19: Pertussis resurgence mini-project

**Topics:**  
A topical mini-project using web-scraping, JSON based APIs and advanced dplyr and ggplot to investigate brand new datasets associated with pertussis cases and longitudinal RNA-Seq on the immune response to distinct vaccination strategies.  

**Hands-on Lab:**
- Lab: [Mini-Project]({{ site.baseurl }}/class-material/class19/lab19_sp23.html){:.no-push-state}{:target="_blank"}

**Supporting material:**
- Additional resources: [CDC pertussis tracking data](https://www.cdc.gov/pertussis/surv-reporting/cases-by-year.html){:.no-push-state}{:target="_blank"} / The [CMI-PB resource]( https://www.cmi-pb.org/){:.no-push-state}{:target="_blank"}

**Homework:**   
- Submit your completed PDF lab report to [GradeScope]({{ site.data.authors.corelinks.gradescope }}){:.no-push-state}{:target="_blank"}


<!--


---
<a name="20"></a>
## 20: GitHub pages online portfolio completion plus bonus Bioinformatics in industry session

**Topics:** 
Todayâ€™s lecture and hands-on sessions introduce Git, currently the most popular version control system. We will learn how to perform common operations with Git and RStudio. We will also cover making a public facing GitHub pages portfolio of your bioinformatics work. Project assignment troubleshooting. Discussion of Bioinformatics and genomics career opportunities.     

**Videos:**  
- 20.1 - OPTIONAL: [Git for humans](https://youtu.be/eWxxfttcMts){:.no-push-state}{:target="_blank"}, 
- 20.2 [Introduction to GitHub Pages](https://youtu.be/2MsN8gpT6jY){:.no-push-state}{:target="_blank"} that we will use for building your portfolio website.
- 20.3 [**Live stream interview**](https://youtu.be/UFYeovcGcwY){:.no-push-state}{:target="_blank"} with leading bioinformatics and genomics scientists from industry including **Dr Ali Crawford** (Associate Director, Scientific Research, *Illumina Inc.*), **Dr. Bjoern Peters** (Full Professor and Principal Investigator, *La Jolla Institute*) and **Dr. Ana Grant** (Director of Research Informatics, *Synthetic Genomics Inc.*).


**Supporting material:**
- Lecture Slides: [Large PDF]({{ site.baseurl }}/class-material/class19_F22.pdf){:.no-push-state}{:target="_blank"},  
- Ether-pad: [**Feedback**](https://board.net/p/bimm143_f22){:.no-push-state}{:target="_blank"}.  
 


- Resource for going further: [Happy Git with R](https://happygitwithr.com/){:.no-push-state}{:target="_blank"},  
- DataCamp: [Bioinformatics Extension Track](https://bioboot.github.io/bimm143_W20/class-material/datacamp_extras.pdf){:.no-push-state}{:target="_blank"} 


Or student topic of choice to be selected from those below:

- Biological network analysis
- Cancer genomics
- Unix tips and tricks for Bioinformatics
- Structural Bioinformatics and computational drug design
- Introduction to the tidyverse 
- Writing R packages
- Advanced RMarkdown 
- Creating online work portfolios with GitHub-pages    


 -->
